pkg/apis/infinispan/v1/types_util.go

```diff
@@ -369,3 +371,69 @@ func (ispn *Infinispan) StorageSize() string {
 	}
 	return ""
 }
+
+// AddLabelsForPods adds to the user maps the labels defined for pods in the infinispan CR
+func (ispn *Infinispan) AddLabelsForPods(uMap map[string]string) {
+	addLabelsFor(ispn, "infinispan.org/podTargetLabels", uMap)
+}
+
+// AddLabelsForServices adds to the user maps the labels defined for services and ingresses/routes in the infinispan CR
+func (ispn *Infinispan) AddLabelsForServices(uMap map[string]string) {
+	addLabelsFor(ispn, "infinispan.org/targetLabels", uMap)
+}
+
+// AddOperatorLabelsForPods adds to the user maps the labels defined for pods in the infinispan CR by the operator
+func (ispn *Infinispan) AddOperatorLabelsForPods(uMap map[string]string) {
+	addLabelsFor(ispn, "infinispan.org/operatorPodTargetLabels", uMap)
+}
+
+// AddOperatorLabelsForServices adds to the user maps the labels defined for services and ingresses/routes in the infinispan CR
+func (ispn *Infinispan) AddOperatorLabelsForServices(uMap map[string]string) {
+	addLabelsFor(ispn, "infinispan.org/operatorTargetLabels", uMap)
+}
+
+func addLabelsFor(ispn *Infinispan, target string, uMap map[string]string) {
+	if ispn.Annotations == nil {
+		return
+	}
+	labels := ispn.Annotations[target]
+	for _, label := range strings.Split(labels, ",") {
+		tLabel := strings.Trim(label, " ")
+		if lval := strings.Trim(ispn.Labels[tLabel], " "); lval != "" {
+			uMap[tLabel] = lval
+		}
+	}
+}
+
+// ApplyOperatorLabels applies operator labels to be propagated to pods and services
+// Env vars INFINISPAN_OPERATOR_TARGET_LABELS, INFINISPAN_OPERATOR_POD_TARGET_LABELS
+// must contain a json map of labels, the former will be applied to services/ingresses/routes, the latter to pods
+func (ispn *Infinispan) ApplyOperatorLabels() {
+	labels := os.Getenv("INFINISPAN_OPERATOR_TARGET_LABELS")
+	labelMap := make(map[string]string)
+	if json.Unmarshal([]byte(labels), &labelMap) == nil && len(labelMap) > 0 {
```

Comment:
Why we need json there, not a simple CSV
The number of configuration parameters become more and more. I think it's time to store them inside separate ConfigMap as described in #491

Reply:
json has a well know/defined syntax and avoids troubles with special use cases like: `I want to store a comma in my label value`.

Reply:
Yes, sure, agreed.
But thinks that Infinispan CR labels and annotations must have higher priority than get from env variable

Reply:
Good spot @dmvolod , I'll add an error log